src/GuessTheNumber.java

```diff
@@ -1,21 +1,29 @@
 public class GuessTheNumber {
+    private int tries;
     private final int number;
     private boolean isOver = false;
+    private IOInterface io;
+    int input;
 
     public GuessTheNumber(IOInterface io, int number) {
         this.number = number;
+        this.io = io;
+        this.tries = 1;
     }
 
     public boolean isMatch(int guessed) {
         if (guessed == number) {
-            return true;
+            return isOver = true;
         }
-        return false;
+        else if (tries == 6) {
+            return isOver = true;
+        }
+        return isOver;
     }
 
-    public boolean gameLoop(int guessedNumber) {
-            if (isMatch(guessedNumber)) {
-              isOver = true;
+    public boolean gameLoop() {
+        while (!isMatch(input = io.getInput())) {
```

Comment:
`input = io.getInput()` feels odd again. I'd rather split that variable into its own line.
Worst case, you can try a 
```
do {
input = io.getInput();
...
} while(!isMatch(input))
```